daemon/graphdriver/devmapper/devmapper_test.go

```diff
@@ -79,3 +80,36 @@ func testChangeLoopBackSize(t *testing.T, delta, expectDataSize, expectMetaDataS
 		t.Fatal(err)
 	}
 }
+
+// Make sure devices.Lock() has been release upon return from cleanupDeletedDevices() function
+func TestDevmapperLockReleasedDeviceDeletion(t *testing.T) {
+	driver := graphtest.GetDriver(t, "devicemapper").(*graphtest.Driver).Driver.(*graphdriver.NaiveDiffDriver).ProtoDriver.(*Driver)
+	defer graphtest.PutDriver(t)
+
+	// Call cleanupDeletedDevices() and after the call take and release
+	// devices.Lock(). If lock has not been released, this will hang.
+	driver.DeviceSet.cleanupDeletedDevices()
+
+	//Wait for 5 seconds for goroutine to finish.
+	timer := time.NewTimer(time.Second * 5)
+	doneChan := make(chan bool)
+
+	go testTakeReleaseLock(driver.DeviceSet, doneChan)
+
+	select {
+	case <-timer.C:
```

Comment:
@lk4d4 Will use time.After()

If timer expires, then I have to do unlock(). Otherwise when I call into the driver to Shutdown(), it hangs and whole unit test machinery hangs.